CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
+
+project(CppExtraSamples)
+
+set(CMAKE_CXX_STANDARD 11)
+set(ZIVID_VERSION 1.5.0)
+
+option(USE_EIGEN3 "Enable samples which depend on Eigen 3" ON)
+option(USE_OPENCV "Enable samples which depend on OpenCV" ON)
+option(USE_PCL "Enable samples which depend on Point Cloud Library (PCL)" ON)
+option(USE_VIS3D "Enable samples which depend on the Zivid Cloud Visualizer. These can be hard to get to build on Linux due to a specific Qt dependency" ON)
+
+find_package(Zivid ${ZIVID_VERSION} COMPONENTS Core REQUIRED)
+find_package(PCL 1.2 QUIET)
+find_package(OpenCV 4.0.1 QUIET OPTIONAL_COMPONENTS core highgui)
+
+set(SAMPLES
+    Zivid/CaptureHDRCompleteSettings
+    Zivid/CaptureHDRLoop
+    Zivid/CaptureSavePLY
+    Zivid/ConnectToSerialNumberCamera
+    Zivid/GetCameraIntrinsics
+    Zivid/ReadZDF
+    Zivid/ZDF2PLY
+    ZividPCL/ReadPCLVis3D
+    ZividPCL/CaptureWritePCLVis3D
+    ZividPCL/CaptureFromFileWritePCLVis3D
+    ZividPCL/ZDF2PCD
+    ZividOpenCV/ZDF2OpenCV
```

Comment:
If you add a Sample here that depends on PCL and you forget to add it to the list below then the CI would still pass I guess.

I think the CI should have a list of deps and first build without deps, then rerun cmake and add one and one of the options and build during each iteration. This should add almost nothing to total build time as it will be incremental builds.

Reply:
Yeah that's a potential problem. I've made a note of it in #15 .